tests/test_prep_run_vasp.py

```diff
@@ -0,0 +1,212 @@
+from PrepRunFp.PrepRunFp import PrepRunFp
+upload_packages.append("../PrepRunFp")
+upload_packages.append("./context.py")
+
+default_config = {
+        "prep":{
+            "template_config" : {
+                "image" : default_image,
+            },  
+        },
+        "run":{
+            "template_config" : {
+                "image" : default_image,
+            },  
+        },
+    }
+
+class TestMockedRunVasp(unittest.TestCase):
+    def setUp(self):
+        warnings.simplefilter('ignore', ResourceWarning)
+        self.ntask = 6
+        self.task_list = []
+        for ii in range(self.ntask):
+            work_path = Path("task.%06d"%ii)
+            work_path.mkdir(exist_ok=True,parents=True),
+            for jj in ['INCAR','POTCAR','POSCAR','KPOINTS']:
+                (work_path/jj).write_text(f'{jj} in task.%06d'%ii)
+            self.task_list.append(work_path)
+
+    def check_run_vasp_output(
+        self,
+        task_name : str,
+    ):
+        cwd = os.getcwd()
+        os.chdir(task_name)
+        fc = []
+        for ii in ['POSCAR','INCAR']:
+            fc.append(Path(ii).read_text())
+        self.assertEqual(fc,Path(Path('my_backward')/'my_log').read_text().strip().split('\n'))
+        self.assertTrue(Path(Path('my_backward')/'POSCAR').is_file())
+        self.assertTrue(Path(Path('my_backward')/'POTCAR').is_file())
+        fc = []
+        fc.append(f"This is POSCAR which users need in {task_name}")
+        fc.append(f"KPOINTS in {task_name}")
+        self.assertEqual(Path(Path('my_backward')/'POSCAR').read_text().strip().split('\n'),fc)       
+        os.chdir(cwd) 
+
+    def tearDown(self):
+        for ii in range(self.ntask):
+            work_path = Path("task.%06d" % ii)
+            if work_path.is_dir():
+                shutil.rmtree(work_path)
+
+    def test(self):
+        self.task_list_str = [str(ii) for ii in self.task_list]
+        for ii in range(self.ntask):
+            ip = OPIO({
+                'task_name' : self.task_list_str[ii],
+                'task_path' : self.task_list[ii],
+                'log_name' : "my_log",
+                'backward_dir_name' : "my_backward",
+                "config" : {},
+                "optional_artifact" : {},
+                "backward_list" : ['POSCAR','POTCAR'],
+                "optional_input" : {},
+            })
+            op = MockedRunVasp()
+            out = op.execute(ip)
+            self.assertEqual(out['backward_dir'] , Path(Path('task.%06d'%ii)/'my_backward'))
+            self.assertTrue(Path(out['backward_dir'] / 'my_log').is_file())
+            self.check_run_vasp_output(self.task_list_str[ii])
+
+@unittest.skipIf(skip_ut_with_dflow, skip_ut_with_dflow_reason)
+class TestPrepRunVasp(unittest.TestCase):
+    def setUp(self):
+        warnings.simplefilter('ignore', ResourceWarning)
+        self.ntasks = 3
+        self.confs = [Path(Path('confs')/'data.000'),Path(Path('confs')/'data.001')]
```

Comment:
Do you really needs the conf files for the testing purpose?

Reply:
I have deleted the conf files.